### ISSUE TITLE
Python 3.10 typehint Type | None

### ISSUE THREAD

**Issue by emdzejbi** [PyCQA/pylint#7871]
### Bug description

`def func() -> str | None: pass
`
returns foo.py:1:14: E1131: unsupported operand type(s) for | (unsupported-binary-operation)

since python 3.10, **Type | None** is preferred way (Optional[Type] in previous releases), therefore this error is false positive

### Command used

```shell
pylint foo.py
```

### Pylint output

```shell
foo.py:1:14: E1131: unsupported operand type(s) for | (unsupported-binary-operation)
```

### Expected behavior

No error reported.

### Pylint version

```shell
pylint 2.15.7
astroid 2.12.13
Python 3.10.4
```

**Comment by cdce8p** [comment]
Please check if `py_version` is set in your config or try setting it to `3.10`.

**Comment by emdzejbi** [comment]
oh sorry, I missed that. With py-version=3.10 it works as expected
